src/ref_db.h

```diff
@@ -0,0 +1,33 @@
+#ifndef GODOT_JVM_REF_DB_H
+#define GODOT_JVM_REF_DB_H
+
+#include <cstdint>
+#include "core/hash_map.h"
+#include "core/reference.h"
+#include "core/list.h"
+#include "core/os/mutex.h"
+
+class RefDB {
+private:
+    RefDB();
+    RefDB(RefDB const &other) = delete;
+    RefDB& operator=(const RefDB&)= delete;
+    ~RefDB() = default;
+
+    struct RefIndex{
+        uint64_t index = 0;
+        uint32_t counter = 0;
+    };
+
+    Mutex* mut;
+    HashMap<uint64_t, RefIndex> ref_map;
```

Comment:
If I understand correctly, it is a map of pointer to RefIndex, so key type should be `uintptr_t`.

Reply:
I did before, and it worked on Windows but seems that MacOs doesn't like this and the CI fails. So I changed it back to uint64-t.
Godot is supposed to handle those types as key:
https://github.com/godotengine/godot/blob/e57f6e2cca90d4c9e956fe1a7ddb3ca086521b42/core/templates/hashfuncs.h#L151